### DISCUSSION TITLE
Как отправить сразу много файлов по почте одним письмом

### DISCUSSION THREAD

**Discussion by Vadimgrig2012** [totumonline/totum-ru-issues-and-discussions#270] [Как это сделать?] [answered]
Всем добрый день. Вопрос заключается в следующем: я хочу по нажатию на кнопку отправить пачку файлов пользователю (отбор конкретных файлов по условию).
И тут немного не понятно: мне взять список или асс.массив? Если асс.массив, то судя по оф доке (может конечно я не правильно понял) нужно вытащить список имен файлов из базы (как????), вложить их в поле field в rowCreate, и далее прировнять их к списку имен, которыми будут подписаны файлы в письме? В общем прошу вашей помощи.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
В общем там у нас была ошибка. Мы исправили и для вашего Cloud обновили.


Отдельный файл отправляется так:

```
= : emailSend(to: "your_email@gmail.com"; title: "test title"; body: "email with files"; files: $files)

files: rowCreate(field: "some_file.md" = #file[0][file])
```

Чтоб несколько по идее можно так:

```
= : emailSend(to: "your_email@gmail.com"; title: "test title"; body: "email with files"; files: $files)

files: rowCreate(field: "some_file_1.md" = #felel[0][file]; field: "some_file_2.md" = #file[0][file])
```

Но это неудобно собирать, поэтому с текущей версии `3.7.49.1` можно передать `rowList` с ключами `name` и `file` (такими же как у поля `file`).

**Вот так:**

```
= : emailSend(to: "your_email@gmail.com"; title: "test title"; body: "email with files"; files: #file)
```

<img width="881" alt="Снимок экрана 2022-11-21 в 19 51 49" src="https://user-images.githubusercontent.com/55755565/203114209-a7f51e25-c083-494f-aa6a-60dc4bae5813.png">
